### PR TITLE
fix(module:tooltip): replace public api with more specific ones

### PR DESCRIPTION
**Pull request by qiurenbo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #5365

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/popconfirm/popconfirm.spec.ts [hunk 1]
```diff
@@ -111,6 +111,20 @@ describe('NzPopconfirm', () => {
     expect(component.confirm).toHaveBeenCalledTimes(1);
     expect(component.cancel).toHaveBeenCalledTimes(0);
   }));
+
+  it('should support changing visibility programmatically', fakeAsync(() => {
```
Comment (wendellhu95) [inline components/popconfirm/popconfirm.spec.ts:115]: Shared logic among tooltip, popover and popconfirm should be tested in tooltip for only once.

File: components/popover/doc/index.zh-CN.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ import { NzPopoverModule } from 'ng-zorro-antd/popover';
 | `[nzPopoverTrigger]` | 触发行为，为 `null` 时不响应光标事件 | `'click' \| 'focus' \| 'hover' \| null` | `'hover'` |
 | `[nzPopoverPlacement]` | 气泡框位置 | `'top' \| 'left' \| 'right' \| 'bottom' \| 'topLeft' \| 'topRight' \| 'bottomLeft' \| 'bottomRight' \| 'leftTop' \| 'leftBottom' \| 'rightTop' \| 'rightBottom'` | `'top'` |
 | `[nzPopoverOrigin]` | 气泡框定位元素 | `ElementRef` | - |
-
+| `[nzPopoverVisible]` | 显示隐藏气泡框 | `boolean` | `false` |
+| `(nzPopoverVisibleChange)` | 显示隐藏的回调 | `EventEmitter<boolean>` | - |
```
Comment (wendellhu95) [inline components/popover/doc/index.zh-CN.md:32]:
```suggestion
| `(nzPopoverVisibleChange)` | 显示隐藏的事件 | `EventEmitter<boolean>` | - |
```

File: components/tooltip/base.ts [hunk 3]
```diff
@@ -122,46 +122,46 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
...
-      this.registerTriggers();
-    }
-
-    if (this.component) {
-      this.updateChangedProperties(changes);
-    }
+  warnDeprecationByChanges(changes: SimpleChanges): void {
+    const getWarnDeprecation = (property: string): string => {
+      return `'${property}' of 'nz-tooltip' is deprecated and will be removed in 10.0.0.
+      Please use 'nzTooltipTitle' instead. The same with 'nz-popover' and 'nz-popconfirm'.`;
```
Comment (wendellhu95) [inline components/tooltip/base.ts:128]: This should differentiate by `property`.

**Comment by wendellhu95** [comment]
LGTM. Thanks!

**Comment by qiurenbo** [comment]
> LGTM. Thanks!

:satisfied:

**Comment by wendellhu95** [comment]
Could you please remove the old `nzVisible` property from the doc? When this new API is introduced by the PR, the old should be considered as _deprecated_.

**Comment by qiurenbo** [comment]
@wendellhu95 OK. I will finish the rest of the work.

**Pull request merged**